h2o-hadoop/scripts/Jenkins/Jenkinsfile

```diff
@@ -0,0 +1,58 @@
+#! /usr/bin/groovy
+
+def AGENT_LABEL = 'mr-0xc10'
+
+pipeline {
+  agent none
+
+  parameters {
+      string(name: 'version', description: 'CDH or HDP version')
+      choice(
+              choices: 'cdh\nhdp',
+              description: 'Hadoop Distribution',
+              name: 'distribution')
+  }
+
+  options {
+    ansiColor('xterm')
+    timestamps()
+    timeout(time: 60, unit: 'MINUTES')
+    // lets keep 33 logs, because currently we test against 11 various hadoop versions
+    buildDiscarder(logRotator(numToKeepStr: '33'))
+  }
+
+  stages {
+    stage('Execute Tests') {
+      agent {
+        dockerfile {
+          label AGENT_LABEL
+          filename "h2o-hadoop/scripts/docker/${params.distribution}/Dockerfile"
+          additionalBuildArgs """--build-arg VERSION=${params.version} \
+            --build-arg PATH_PREFIX=h2o-hadoop/scripts/docker/${params.distribution} \
+            --build-arg DEFAULT_USER_UID=\$(id -u)"""
+          args """-u root \
+            -v \${WORKSPACE}:/home/h2o/h2o-3 \
+            -v \${WORKSPACE}/h2o-hadoop/scripts/docker/common/custom_startup/:/startup/
+            -e H2O_DOWNLOAD_TIMEOUT=2m"""
+          reuseNode true
+        }
+      }
+      steps {
+        sh "/usr/sbin/startup.sh"
+      }
+    }
+  }
+
+  post {
+    failure {
+      emailext (
```

Comment:
We have utilities for notification in common jenkins library - see https://github.com/h2oai/h2oai/blob/dev/Jenkinsfile#L259
and
https://github.com/h2oai/h2o-jenkins-pipeline-lib/blob/master/vars/sendNotifications.groovy

You can contribute to Jenkins lib with PR.